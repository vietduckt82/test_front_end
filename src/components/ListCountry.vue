<template>
  <div>
    <h3>List Country</h3>
    <div class="search-page">
      <div class="seach">
        <img :src="getSeachIcon" />
        <input
          class="pd-input"
          type="text"
          name="seach"
          placeholder="Search Country Name"
          v-model="filterObj.name"
        />
      </div>
    </div>

    <table class="list-country">
      <tbody v-if="getCountryfilter && getCountryfilter.length">
        <tr>
          <th>Country</th>
          <th>NewConfirmed</th>
          <th>NewDeaths</th>
          <th>NewRecovered</th>
          <th>TotalConfirmed</th>
          <th>TotalDeaths</th>
          <th>TotalRecovered</th>
          <th>Action</th>
        </tr>
        <tr v-for="post in getCountryfilter" :key="post.id">
          <td>{{ post.Country }}</td>
          <td>{{ post.NewConfirmed }}</td>
          <td>{{ post.NewDeaths }}</td>
          <td>{{ post.NewRecovered }}</td>
          <td>{{ post.TotalConfirmed }}</td>
          <td>{{ post.TotalDeaths }}</td>
          <td>{{ post.TotalRecovered }}</td>
          <td>
            <router-link :to="`/country/${post.Slug}`">Detail</router-link>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</template>
<script>
import SeachIcon from "@/assets/icon/search.png";
import { cloneDeep } from "lodash";

export default {
  props: {
    data: {
      type: Array,
    },
  },
  data() {
    return {
        filterObj: {
            name: null,
        },
    };
  },
  methods: {},
  computed: {
    getSeachIcon() {
      return SeachIcon;
    },
    getCountryfilter() {
        console.log(this.data);
      let listData = cloneDeep(this.data);
      if (this.filterObj.name) {
        listData = listData.filter((item) => {
          if (
            item.Country.toLowerCase().includes(
              this.filterObj.name.toLowerCase()
            )
          ) {
            return item;
          }
        });
      }
      return listData;
    },
  },
};
</script>