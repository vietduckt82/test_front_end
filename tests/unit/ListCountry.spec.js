import { mount  } from "@vue/test-utils";
import ListCountry from "@/components/ListCountry.vue";

describe("ListCountry.spec.js", () => {
    let cmp;
    beforeEach(() => {
		cmp = mount(ListCountry, {
			// Beaware that props is overriden using `propsData`
        propsData: {
          data: [
            {
              Active: 0,
              City: "",
              CityCode: "",
              Confirmed: 0,
              Country: "Viet Nam",
              CountryCode: "",
              Date: "2020-01-22T00:00:00Z",
              Deaths: 0,
              Lat: "0",
              Lon: "0",
              Province: "",
              Recovered: 0
            },
          ],
        },
      });
    });
    
    it('has received ["Cat"] as the message property', () => {
      expect(cmp.vm.data).toEqual([
        {
          Active: 0,
            City: "",
            CityCode: "",
            Confirmed: 0,
            Country: "Viet Nam",
            CountryCode: "",
            Date: "2020-01-22T00:00:00Z",
            Deaths: 0,
            Lat: "0",
            Lon: "0",
            Province: "",
            Recovered: 0
        },
      ]);
    });
    it("has the expected html structure", () => {
      expect(cmp.element).toMatchSnapshot();
    });
    it("is a Home component", () => {
      expect(cmp.is(ListCountry)).toBe(true);
    });
    // Vue instance
    it("Home is a vue instances", () => {
      expect(cmp.isVueInstance()).toBe(true);
    });
});