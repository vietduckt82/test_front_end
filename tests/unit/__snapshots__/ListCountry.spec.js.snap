// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCountry.spec.js has the expected html structure 1`] = `
<div>
  <h3>
    List Country
  </h3>
   
  <div
    class="search-page"
  >
    <div
      class="seach"
    >
      <img
        src=""
      />
       
      <input
        class="pd-input"
        name="seach"
        placeholder="Search Country Name"
        type="text"
      />
    </div>
  </div>
   
  <table
    class="list-country"
  >
    <tbody>
      <tr>
        <th>
          Country
        </th>
         
        <th>
          NewConfirmed
        </th>
         
        <th>
          NewDeaths
        </th>
         
        <th>
          NewRecovered
        </th>
         
        <th>
          TotalConfirmed
        </th>
         
        <th>
          TotalDeaths
        </th>
         
        <th>
          TotalRecovered
        </th>
         
        <th>
          Action
        </th>
      </tr>
       
      <tr>
        <td>
          Viet Nam
        </td>
         
        <td>
          
        </td>
         
        <td>
          
        </td>
         
        <td>
          
        </td>
         
        <td>
          
        </td>
         
        <td>
          
        </td>
         
        <td>
          
        </td>
         
        <td>
          <router-link
            to="/country/undefined"
          >
            Detail
          </router-link>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
